src/foam/lib/json/StringParser.java

```diff
@@ -19,7 +19,7 @@ public StringParser() {
   public PStream parse(PStream ps, ParserContext x) {
     delim_ = ps.head();
     if ( delim_ != '"' && delim_ != '\'' ) {
-      return null;
+      return (new Literal("null", "")).parse(ps, x);
```

Comment:
This will create a new literal every time we try to parse a string.  I also don't think we should be transforming a literal null into empty string.  If some JSON contains "foo": null, then I expect later getFoo() on my object to return null.

At the very least it's not the responsibility of the StringParser to make these decisions.